### PR TITLE
[14.0][MIG]Migrated account_invoice_refund _reason

### PR DESCRIPTION
**Pull request by SilvioC2C**
Copies #905 and implements it (including change requests that were introduced in that PR's reviews).

NB: I tried to open a PR against the original branch from which PR #905 was opened, but GitHub does not let me do that, so I cannot update it.

### CODE REVIEW

**Review by tarbib — APPROVED** [review]
https://user-images.githubusercontent.com/2255433/142992957-4346a0cb-0419-47d5-aed3-0ffa751c0179.mp4

Functional tests ok :+1:

File: account_invoice_refund_reason/i18n/account_invoice_refund_reason.pot [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Translation of Odoo Server.
```
Comment (i-vyshnevska) [inline account_invoice_refund_reason/i18n/account_invoice_refund_reason.pot:1]: we removing pot files as they obsolete from some version

File: account_invoice_refund_reason/tests/test_account_invoice_refund_reason.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            {
+                "reason": "no reason",
+                "date": datetime.date.today() + datetime.timedelta(days=1),
+                "refund_method": "refund",
+            }
+        )
+
+    def _play_reason_onchange(self):
+        self.refund_wizard.reason_id = self.reason
+        self.refund_wizard._onchange_reason_id()
```
Comment (i-vyshnevska) [inline account_invoice_refund_reason/tests/test_account_invoice_refund_reason.py:69]: FTR (no need to fix if we don't have time here) if we testing UI there is Form class that guarantees to us to play onchanges in the same way as UI
https://github.com/odoo/odoo/blob/da787012c1b6774d01f25c9dcc42d6161730ed16/addons/l10n_ch/tests/test_vendor_bill_isr.py#L59

**Review by i-vyshnevska — APPROVED** [review]
Translations commit can be squashed together

Screenshot

**Comment by yankinmax** [comment]
@max3903 can you pls take a look?

**Comment by ChrisOForgeFlow** [comment]
@SilvioC2C can you rebase code please?

**Comment by SilvioC2C** [comment]
@ChrisOForgeFlow done :muscle:

**Comment by MiquelRForgeFlow** [comment]
please cherry-pick https://github.com/OCA/account-invoicing/commit/f43f1e703c71e0491b47d3bd2eae5ff32cf76bf7

**Comment by AaronHForgeFlow** [comment]
Hi @SilvioC2C , this PR was superseded here: https://github.com/OCA/account-invoicing/pull/1138 and the CI is fixed on that one. Now, you can choose between closing your PR and accept the supersede, or keep using your PR and fix it. Just let the community know what you choose.

**Comment by SilvioC2C** [comment]
@AaronHForgeFlow thanks for the news, I'll close this PR and review #1273 instead!

**Pull request closed without merging**